src/gui/qgsprojectionselectionwidget.h

```diff
@@ -100,6 +101,46 @@ class GUI_EXPORT QgsProjectionSelectionWidget : public QWidget
      */
     static QString crsOptionText( const QgsCoordinateReferenceSystem &crs ) SIP_SKIP;
 
+    /**
+     * Returns TRUE if the widget will show a warning to users when they select a CRS which has
+     * low accuracy.
+     *
+     * \see setShowAccuracyWarnings()
+     * \since QGIS 3.20
+     */
+    bool showAccuracyWarnings() const;
+
+    /**
+     * Sets whether the widget will \a show warnings to users when they select a CRS which has
+     * low accuracy.
+     *
+     * \see showAccuracyWarnings()
+     * \since QGIS 3.20
+     */
+    void setShowAccuracyWarnings( bool show );
+
+    /**
+     * Sets the original source \a ensemble datum name.
+     *
+     * If set, CRS accuracy warnings will not be shown when the selected CRS in the widget has a matching
+     * ensemble datum, regardless of the ensemble's accuracy.
+     *
+     * \see sourceEnsemble()
+     * \since QGIS 3.20
+     */
+    void setSourceEnsemble( const QString &ensemble );
```

Comment:
My remark is perhaps not relevant as I'm not super familiar with this widget, but I'm wondering if that couldn't be something automatically done by the existing setCrs() method ?

Reply:
Hmmmm, good question. I'd prefer to keep them separate as I think there's use cases when we might have a source layer in some crs yet want to show a different crs as default in the widget -- especially if we want the widget to default to 4326/3857 for some uses.